pmtiles/bucket_test.go

```diff
@@ -206,3 +213,54 @@ func TestFileShorterThan16K(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, 3, len(data))
 }
+
+func TestSetProviderEtagAws(t *testing.T) {
+	var awsV1Req s3.GetObjectInput
+	assert.Nil(t, awsV1Req.IfMatch)
+	asFunc := func(i interface{}) bool {
+		v, ok := i.(**s3.GetObjectInput)
+		if ok {
+			*v = &awsV1Req
+		}
+		return true
+	}
+	setProviderEtag(asFunc, "123")
+	assert.Equal(t, aws.String("123"), awsV1Req.IfMatch)
+}
+
+func TestSetProviderEtagAzure(t *testing.T) {
+	var azOptions azblob.DownloadStreamOptions
+	assert.Nil(t, azOptions.AccessConditions)
+	asFunc := func(i interface{}) bool {
+		v, ok := i.(**azblob.DownloadStreamOptions)
+		if ok {
+			*v = &azOptions
+		}
+		return ok
+	}
+	setProviderEtag(asFunc, "123")
+	assert.Equal(t, azcore.ETag("123"), *azOptions.AccessConditions.ModifiedAccessConditions.IfMatch)
+}
```

Comment:
Could you have a similar test for google cloud?

Reply:
I wasn't able to since conditions is not a public field of the gcp interface (is write-only through the If function)

Reply:
Ok got it. This is probably as good as we can get on testing short of standing up a mock implementation of the provider's API 👍